cranelift/interpreter/src/step.rs

```diff
@@ -824,8 +824,28 @@ where
         )?),
         Opcode::SwidenLow => unimplemented!("SwidenLow"),
         Opcode::SwidenHigh => unimplemented!("SwidenHigh"),
-        Opcode::UwidenLow => unimplemented!("UwidenLow"),
-        Opcode::UwidenHigh => unimplemented!("UwidenHigh"),
+        Opcode::UwidenLow => {
+            let new_type = ctrl_ty.merge_lanes().unwrap();
+            let mut new_vec = SimdVec::new();
+            let mut arg0 = extractlanes(&arg(0)?, ctrl_ty.lane_type())?;
+            arg0.truncate(new_type.lane_count() as usize);
+            for lane in arg0 {
+                let lane = lane.convert(ValueConversionKind::ZeroExtend(new_type.lane_type()))?;
+                new_vec.push(lane);
+            }
```

Comment:
Could we write this with iterator methods (similar to @afonso360's comment in #3314)? Something like:

`let new_vec = extractlanes(...).into_iter().take(new_type.lane_count).map(|lane| lane.convert(...)).collect::<Result<Vec<_>>()?`

(note that there is a very convenient trait impl in libstd that lets one `collect` individual `Result<T>` items into `Result<Vec<T>>` which takes care of the try operator in the loop!)